<launch>
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi, ball_waffle_pi, ball2_waffle_pi]"/>
  <arg name="x_pos" default="1"/>
  <arg name="y_pos" default="3.5"/>
  <arg name="z_pos" default="0.0"/>


  <param name="range_min" type="double" value=".12" />
  <param name="range_max" type="double" value="3.5" />
  <param name="a_min" type="double" value="0" />
  <param name="a_max" type="double" value="6.28" />
  <param name="a_inc" type="double" value=".0175" />
  <param name="scan_time" type="double" value="0" />

<!--    <node name="meriwether_broadcaster" pkg="navigate" type="broadcast.py" respawn="false" output="screen" >
    <param name="turtle" type="string" value="meriwether" />
  </node>  -->

  <node name="fetcher" pkg="fetch" type="fetcher.py" respawn="false" output="screen" >
  </node>



  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find fetch)/worlds/challenge-world-t1"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
<!--     <arg name="range_min" value=".1"/> -->
  </include>


  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />

  <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf" args="-urdf -model turtlebot3_$(arg model) -x $(arg x_pos) -y $(arg y_pos) -z $(arg z_pos) -param robot_description" />


  <!-- JOINT CONTROLLER STUFF -->
  <rosparam file="$(find fetch)/joint_config.yaml" ns="/control" command="load" />
  <node name="controller_spawner" pkg="controller_manager" type="spawner" ns="/control" respawn="false" output="screen" args="base_to_arm_joint_position_controller joint_state_controller"/> 


<!-- 
  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
    <param name="publish_frequency" type="double" value="50.0" />
    <param name="tf_prefix" value="meriwethertf"/>
  </node>
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find turtlebot3_gazebo)/rviz/turtlebot3_gazebo_model.rviz"/> -->
</launch>
